.github/workflows/r-extended.yml

```diff
@@ -0,0 +1,200 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: R (extended)
+
+on:
+  # Run weekly
+  schedule:
+    - cron: '5 0 * * 0'
+
+  pull_request:
+    branches:
+      - main
+    paths:
+      - ".github/workflows/r-extended.yml"
+
+permissions:
+  contents: read
+
+jobs:
+
+  # Runs R CMD check in the same way CRAN might (i.e., without any test databases,
+  # using oldrel, release, devel.
+  check-for-cran:
+    name: "cran - ${{ matrix.pkg }} / ${{ matrix.os }} / R-${{ matrix.rversion}}"
+    runs-on: ${{ matrix.os }}-latest
+
+    strategy:
+      matrix:
+        rversion: [oldrel, release, devel]
+        os: [macOS, windows, ubuntu]
+        pkg: [adbcdrivermanager, adbcsqlite, adbcpostgresql, adbcflightsql, adbcsnowflake]
+      fail-fast: false
+
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Any secrets should not be set globally as envvars (neither workflow nor job wide) and only set in the step they are used. That way potentially compromised actions can't grab them. Not essential here as the permissions are set to read only.